app/controllers/events/view/speakers/list.js

```diff
@@ -90,8 +90,9 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
   }
 
   @action
-  viewSpeaker(id) {
-    this.transitionToRoute('events.view.speakers.edit', id);
+  viewSpeaker(speaker_id, event) {
+    const event_id = event.router.location.location.pathname.split('/')[2];
```

Comment:
Use speaker object to get event ID

Reply:
Do I have to find record of speaker in this action with the help of speaker_id and extract event id?